### PR TITLE
Fix after testnet updates

### PR DESCRIPTION
**Pull request by ketola**

### CODE REVIEW

File: jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetEventsByEventAccessPath.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                .appendInt(STRUCT_TAG_TYPE_PARAMS_LENGTH)
+                .toByteSequence()
+                .toArray();
+
+        byte[] salt = digestSHA3.digest("StructTag::libra_types::language_storage@@$$LIBRA$$@@".getBytes());
+        byte[] saltAndStructTag = new byte[salt.length + serializedStructTag.length];
+        System.arraycopy(salt, 0, saltAndStructTag, 0, salt.length);
+        System.arraycopy(serializedStructTag, 0, saltAndStructTag, salt.length, serializedStructTag.length);
+        byte[] structTagHash = digestSHA3.digest(saltAndStructTag);
+
```
Comment (ice09) [inline jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetEventsByEventAccessPath.java]: All fine, just realized that I saw this pattern very often when dealing with Libra:
```
        byte[] first = digestSHA3.digest("WHATEVER".getBytes());
        byte[] firstAndLast = new byte[first.length + last.length];
        System.arraycopy(first, 0, firstAndLast, 0, first.length);
        System.arraycopy(firstAndLast, 0, firstAndLast, first.length, firstAndLast.length);
        byte[] firstAndLastHash = digestSHA3.digest(firstAndLast);
```
maybe it makes sense to "utilize" this later on.

Comment (ketola) [reply jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetEventsByEventAccessPath.java]: I created a new Hash class and moved the duplicated code there

File: jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetEventsByEventAccessPath.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import types.GetWithProof.RequestItem;
+
+@Value.Immutable
+public abstract class GetEventsByEventAccessPath {
+
+    private static final int STRUCT_TAG_TYPE_PARAMS_LENGTH = 0;
+    private static final String STRUCT_TAG_MODULE = "T";
+    private static final String STRUCT_TAG_ADDRESS = "LibraAccount";
+    private static final String STRUCT_TAG_ACCOUNT_ADDRESS = "0000000000000000000000000000000000000000000000000000000000000000";
+    private static final byte RESOURCE_TAG = (byte) 1;
```
[the remark this answers is not in the document]
Comment (ketola) [reply jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetEventsByEventAccessPath.java]: Fixed

File: jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetEventsByEventAccessPath.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    public abstract long getStartEventSequenceNumber();
+
+    public abstract boolean isAscending();
+
+    public RequestItem toGrpcObject() {
+        return RequestItem.newBuilder()
+                .setGetEventsByEventAccessPathRequest(GetEventsByEventAccessPathRequest.newBuilder()
+                        .setAccessPath(AccessPath.newBuilder()
+                                .setAddress(ByteString.copyFrom(getAccountAddress().getByteSequence().toArray()))
```
[the remark this answers is not in the document]
Comment (ketola) [reply jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/GetEventsByEventAccessPath.java]: Fixed

**Pull request merged**